### PR TITLE
fix(react-scheduler): move firstDayOfWeek prop from views to Scheduler

### PR DESCRIPTION
**Pull request by AryamnovEugeniy**
BREAKING CHANGES:

The `MonthView` and `WeekView` plugins now do not have firstDayOfWeek property. It was moved to the `Scheduler` plugin.

```diff
<Scheduler
+  firstDayOfWeek={firstDayOfWeek}
>
  <MonthView
-    firstDayOfWeek={firstDayOfWeek}
  >
  <WeekView
-    firstDayOfWeek={firstDayOfWeek}
  >
   ...
</Scheduler>
```

### CODE REVIEW

File: packages/dx-react-scheduler/src/plugins/week-view.test.tsx [hunk 1]
```diff
@@ -126,7 +126,10 @@ describe('Week View', () => {
       };
       const tree = mount((
         <PluginHost>
-          {pluginDepsToComponents(defaultDeps)}
+          {pluginDepsToComponents({
+            ...defaultDeps,
+            getter: { ...defaultDeps.getter, firstDayOfWeek },
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/src/plugins/week-view.test.tsx]: Why we don't add this getter into `defaultDeps` variable?

File: packages/dx-react-scheduler/src/plugins/scheduler-core.tsx [hunk 2]
```diff
@@ -7,6 +7,10 @@ import { SchedulerProps } from '../types';
 import { memoize } from '@devexpress/dx-core';
 
 class SchedulerCoreBase extends React.PureComponent<SchedulerProps> {
+  static defaultProps: Partial<SchedulerProps> = {
+    firstDayOfWeek: 0,
+  };
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/src/plugins/scheduler-core.tsx]: We have defined default properties in the `scheduler.tsx` file.

File: packages/dx-react-scheduler/src/types/scheduler/scheduler.types.ts [hunk 3]
```diff
@@ -23,4 +23,6 @@ export interface SchedulerProps {
    * the height equals that of the container component.
    * **/
   height: number | 'auto';
+  /** A number between 0 (Sunday) and 6 (Saturday) that specifies the first day of the week. */
+  firstDayOfWeek?: number;
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/src/types/scheduler/scheduler.types.ts]: it's should be required property.

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ Name | Type | Default | Description
 data | Array&lt;[AppointmentModel](#appointmentmodel)&gt; | | An array of appointment data objects.
 height | number &#124; `auto` | `auto` | The scheduler's height. If the value is `auto`, the height equals that of the container component.
 locale | string &#124; Array&lt;string&gt; | 'en-US' | The locale according to which dates should be formatted.
+firstDayOfWeek? | number | A number between 0 (Sunday) and 6 (Saturday) that specifies the first day of the week.
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/docs/reference/scheduler.md]: Add the `Default` column to your message.

**Comment by AryamnovEugeniy** [comment]
FPR

**Pull request merged**